### DISCUSSION TITLE
[BUG] Start menu style Windows 10 Full Screen

### DISCUSSION THREAD

**Discussion by hiranokite** [valinet/ExplorerPatcher#933] [General]
Windows Version: 21H2 Build [22000.527]
ExplorerPatch Version: [22000.493.42.18]

I upgraded from 10 to 11 just to test the new Windows 10 start menu, but when i apply the Windows 10 style, I could access the menu only one time after pressing the WIN LOGO on keyboard (if click on start button on taskbar not happens), full screen 10 start menu show up one time after a delay of 1-2 seconds, but after that i cant access the menu again without apply the 11 style and back to 10 style.

After some tests i discovered that it seems a bug with FULL SCREEN START MENU, as we no longer have the option to change the view mode, the only alternative I had was to create the key on REGEDIT to force the menu always in small mode.

1. navigate to HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows
2. new key name Explorer
3. new DWORD-32 ForceStartSize set to 1 to small start menu (2 = full screen, to reproduce this bug)
4. restart explorer or reboot

hope this can be fixed.

**Comment by valinet** [comment]
Yeah, I haven’t tested that, the solution for now is to use it as a menu. Unset that policy and it will display as a menu by default.

**Reply by hiranokite** [reply]
not in my case, i upgraded from windows 10 with full screen menu enabled on my user, so i can't get the menu to work properly without setting this policy.

**Reply by valinet** [reply]
Okay, yeah, then set the policy accordingly to your case. There probably is some setting in the registry other than that that controls whether the menu is displayed full screen or not as well.

**Reply by hiranokite** [reply]
Just to complete, windows 10 start menu "Show More Tiles" is working, after the upgrade from 10, but i don't know what registry key enable.

Thanks for that amazing tool!

![sm](https://user-images.githubusercontent.com/5361459/155841953-f11eb62b-2a55-49c9-807e-58629e96e240.png)

**Reply by valinet** [reply]
Yeah, me neither. Maybe all these should be documented and added to the Properties GUI probably.

**Reply by valinet** [reply]
I have implemented support for full screen Start menu in latest pre-release (42.23).

**Reply by hiranokite** [reply]
thank you for this huge update, now only need a way to enable show more tiles, to complete the menu, i cant find the registry key to do it.

**Reply by valinet** [reply]
Me neither, I was looking for it, I will boot up a Windows 10 VM, watch the registry and see if that works here. Google isn't of much help. Since you say it works (from your screenshot), something clearly queries it.

**Reply by valinet** [reply]
And you're welcome :D

**Reply by hiranokite** [reply]
i tried to use regshot, but no success, hope you can find

**Reply by valinet** [reply]
Yeah, another Microsoft trademark, it seems to be yet another convoluted way to set a setting. Pff...

**Reply by hiranokite** [reply]
I managed to enable it with a hack 😅, now i'm trying to figure out how to do this correctly

**Reply by hiranokite** [reply]
![image](https://user-images.githubusercontent.com/5361459/156208187-6b33467e-7ea0-4b5f-93c6-3363ee38edce.png)

**Reply by valinet** [reply]
What hack? I found where it is located: `Computer\HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\CloudStore\Store\Cache\DefaultAccount\$de${c34c7785-5518-473a-91c2-252e95bfe3c6}$start.tilegrid$windows.data.curatedtilecollection.tilecollection\Current\Data`. It's byte 26. `0c` (12) means 3 columns, `10` (16) means 4 columns, aka "Show more tiles". Change that byte and restart File Explorer.

Idk if this thing: `$de${c34c7785-5518-473a-91c2-252e95bfe3c6}$start.tilegrid$windows.data.curatedtilecollection.tilecollection` is unique to my install or not. Anyway, I will install a hook in `StartMenuExperienceHost.exe` so that when it reads that value, I override it with a value that can be set via the Properties GUI.

**Reply by hiranokite** [reply]
i just exported the menu layout from 10 with 4 row enabled, and imported to 11

**Reply by valinet** [reply]
Aha, makes sense. Yeah, I will try implementing what I said, hopefully that will turn into a nice option in the GUI that will alter that setting and that'd be it.

**Reply by hiranokite** [reply]
`Idk if this thing: $de${c34c7785-5518-473a-91c2-252e95bfe3c6}$start.tilegrid$windows.data.curatedtilecollection.tilecollection is unique to my install or not`

its unique, my vm is that
$de${ad5613bd-63b1-40ab-8ef3-08c1b0c3ee38}$start.tilegrid$windows.data.curatedtilecollection.tilecollection

**Reply by hiranokite** [reply]
`\HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\CloudStore\Store\Cache\DefaultAccount\$de${c34c7785-5518-473a-91c2-252e95bfe3c6}$start.tilegrid$windows.data.curatedtilecollection.tilecollection\Current\Data. It's byte 26. 0c (12) means 3 columns, 10 (16) means 4 columns, aka "Show more tiles". Change that byte and restart File Explorer.`

thanks for the explanation, I was able to enable it easily now

**Reply by valinet** [reply]
Yeah, that `$de${c34c7785-5518-473a-91c2-252e95bfe3c6}$start.tilegrid$windows.data.curatedtilecollection.tilecollection` thing doesn't really matter, as EP will offer option in the GUI to enable/disable this, I match just the `$start.tilegrid$windows.data.curatedtilecollection.tilecollection\Current` part which seems to be unique enough not to get in the way of other things. I am testing it now, seems to work fine.

**Reply by hiranokite** [reply]
nice 😀

**Reply by hiranokite** [reply]
@valinet everything working, thanks for the great work ❤️
